File: src/main/java/uk/co/alt236/apkcompare/comparators/results/CollectionSimilarityEvaluator.java
package uk.co.alt236.apkcompare.comparators.results;

import java.util.Collection;
import java.util.EnumSet;
import java.util.stream.Collectors;

class CollectionSimilarityEvaluator {

    static Similarity evaluate(final Collection<? extends ComparisonResult> items) {
        EnumSet<Similarity> enumSet =
                items
                        .stream()
                        .map(ComparisonResult::getSimilarity)
                        .collect(Collectors.toCollection(() -> EnumSet.noneOf(Similarity.class)));

        if (enumSet.contains(Similarity.DIFFERENT)) {
            return Similarity.DIFFERENT;
        } else {
            return Similarity.IDENTICAL;
        }
    }
}

File: app/src/main/java/uk/co/alt236/apkcompare/app/comparators/results/CollectionSimilarityEvaluator.kt
package uk.co.alt236.apkcompare.app.comparators.results

object CollectionSimilarityEvaluator {

    @JvmStatic
    fun evaluate(items: Collection<ComparisonResult>): Similarity {

        for (result in items) {
            if (result.similarity !== Similarity.IDENTICAL) {
                return result.similarity
            }
        }

        return Similarity.IDENTICAL
    }
}
